src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,124 @@
+import React, {
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type TodoItemProps = {
+  todo: Todo;
+  isLoading: boolean;
+  onDelete: (id: number) => void;
+  onStatusChanged: () => void;
+  updateTodo: (id: number, newTitle: string) => void;
+};
+
+export const TodoItem: React.FC<TodoItemProps> = ({
+  todo: { id, title, completed },
+  isLoading,
+  onDelete,
+  onStatusChanged,
+  updateTodo,
+}) => {
+  const [editMode, setEditMode] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const handleTitleUpdate = useCallback(() => {
+    if (!newTitle.trim()) {
+      onDelete(id);
+    } else if (newTitle !== title) {
+      updateTodo(id, newTitle);
+    }
+
+    setEditMode(false);
+  }, [newTitle]);
```

Comment:
The `handleTitleUpdate` function is missing the necessary dependencies in the dependency array. It should include `id`, `onDelete`, `title`, and `updateTodo`.